src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigRunners.cs

```diff
@@ -100,6 +115,19 @@ public static ISettings GetSettingsFromDirectory(string directory)
         }
 
         /// <summary>
+        /// Creates a settings object utilizing a NuGet configuration file path.
+        /// </summary>
+        /// <param name="filePath"></param>
+        /// <returns></returns>
+        public static ISettings GetSettingsFromFile(string filePath)
+        {
+            var directory = Path.GetDirectoryName(filePath);
+            return NuGet.Configuration.Settings.LoadDefaultSettings(
```

Comment:
That makes sense! I misunderstood how `LoadDefaultSettings` works. I'm editing this to use `LoadSpecificSettings` instead, which appears to do what I actually wanted:
https://github.com/NuGet/NuGet.Client/blob/25d848af193c5dc17530525be8d0952d0335e8f1/src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs#LL346C1-L363C10

I'll take a closer look at the tests I have, because they passed when I was using `LoadDefaultSettings`, and I'm now not sure that they should have passed.